clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -1415,4 +1415,21 @@ public static boolean isBlank(String str) {
         return res;
     }
 
+    /**
+     * Get an array containing all of the {@link Object#toString names} of a given enumerable type.
+     * @param enumClass the enum class; may not be null
+     * @return an array with the names of every value for the enum class; never null, but may be empty
+     * if there are no values defined for the enum
+     */
+    public static String[] enumOptions(Class<? extends Enum<?>> enumClass) {
+        Objects.requireNonNull(enumClass);
+        if (!enumClass.isEnum()) {
+            throw new IllegalArgumentException("Class " + enumClass + " is not an enumerable type");
+        }
+
+        return Stream.of(enumClass.getEnumConstants())
+                .map(Object::toString)
```

Comment:
Is it correct to use `Object::toString` here, rather than `Enum::name`? If so, maybe the javadoc needs adjusting since it's not guaranteed to return the enum value name.

Reply:
I opted for `toString` instead of `name` since the former can be overridden, but the latter can't. The Javadoc does explicitly link to the `toString` method, but that's probably not enough to cancel out the assumption people might make that the text "name" refers to `Enum::name`.

I can tweak it to say "string representations", which is a little clunkier but shouldn't mislead people like "name" might. LMK if you have other thoughts.